dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1266,21 +1318,34 @@ synchronized String evaluateInterval() {
                 cpuHigh = true;
             }
 
+            boolean lowTput = false;
+            if (throughput / poolSize < lowTputThreadsRatio && queueDepth == 0) {
+                lowTput = true;
+                if (tc.isEventEnabled()) {
+                    Tr.event(tc, "low tput flag set: throughput: " + throughput + ", poolSize: "
+                                 + poolSize + ", queueDepth: " + queueDepth,
+                             threadPool);
+                }
+            }
+
             setPoolIncrementDecrement(poolSize);
 
             double forecast = currentStats.getMovingAverage();
-            double shrinkScore = getShrinkScore(poolSize, forecast, throughput, queueDepth, cpuHigh);
-            double growScore = getGrowScore(poolSize, forecast, throughput, queueDepth, cpuHigh);
+            double shrinkScore = getShrinkScore(poolSize, forecast, throughput, queueDepth, cpuHigh, lowTput);
+            double growScore = getGrowScore(poolSize, forecast, throughput, queueDepth, cpuHigh, lowTput);
 
+            // Adjust the poolsize only if one of the scores is both larger than the scoreFilterLevel
+            // and sufficiently larger than the other score. These conditions reduce poolsize fluctuation
+            // which might arise due to a weak or noisy signal from the historical throughput data.
             int poolAdjustment = 0;
-            if (growScore > shrinkScore) {
+            if (growScore > growScoreFilterLevel && growScore > (shrinkScore + growShrinkDiffFilter)) {
```

Comment:
This doesn't look right.  I am thinking something more like:
```
if (Math.abs(growScore - shrinkScore) > growShrinkDiffFilter) {
    if (growScore > growScoreFilterLevel && growScore > shrinkScore) {
        poolAdjustment = poolIncrement;
    } else if (shrinkScore > shrinkScoreFilterLevel && shrinkScore > growScore) {
        poolAdjustment = -poolIncrement;
    }
}
```

Reply:
This is I think functionally equivalent to the prior code, but will in some cases be more efficient, so I agree and changed accordingly.